domaindriven-core/src/DomainDriven/Persistance/Class.hs

```diff
@@ -28,11 +28,36 @@ class ReadModel p where
 type TransactionalUpdate model event m a = (model -> m (model -> a, [event])) -> m a
 
 class ReadModel p => WriteModel p where
+    -- | Hook to call after model has been updated.
+    -- This allows for setting up outgoing hooks in calling out to external systems.
+    -- This is run in asyncly after update is processed.
+    postUpdateHook :: MonadIO m => p -> Model p -> [Stored (Event p)] -> m ()
+
+    -- | How to perform a transactional update, returning what is required
+    -- for postUpdateHook and returning the result to the user
```

Comment:
I find this comment confusing. Maybe rename to something like `transactionalUpdateImpl` to reinforce that this isn't a public facing api?

My proposal is something along

-- | Update the model in a transaction. Note that this is never used directly; runCmd calls transactionalUpdate and makes sure to call postUpdateHook afterwards.

Reply:
sure